src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java

```diff
@@ -0,0 +1,900 @@
+      CConnectorMqttMgr mqttMgr, String mqttTopic, String message, String expectedDeviceId) {
+    return false;
+  }
+
+  /**
+   * Not yet implemented (Unavailable).
+   *
+   * @param mqttMgr Not yet implemented (Unavailable)
+   * @param mqttTopic Not yet implemented (Unavailable)
+   * @param message Not yet implemented (Unavailable)
+   * @param expectedDeviceId Not yet implemented (Unavailable)
+   * @return Not yet implemented (Unavailable)
+   */
+  public static boolean setDeviceProfiles_527(
+      CConnectorMqttMgr mqttMgr, String mqttTopic, String message, String expectedDeviceId) {
+    return false;
+  }
+
+  /**
+   * Not yet implemented (Unavailable).
+   *
+   * @param mqttMgr Not yet implemented (Unavailable)
+   * @param mqttTopic Not yet implemented (Unavailable)
+   * @param message Not yet implemented (Unavailable)
+   * @param expectedDeviceId Not yet implemented (Unavailable)
+   * @return Not yet implemented (Unavailable)
+   */
+  public static boolean updateSoftware_528(
+      CConnectorMqttMgr mqttMgr, String mqttTopic, String message, String expectedDeviceId) {
+    return false;
+  }
+
+  /**
+   * Not yet implemented (Unavailable).
+   *
+   * @param mqttMgr Not yet implemented (Unavailable)
+   * @param mqttTopic Not yet implemented (Unavailable)
+   * @param message Not yet implemented (Unavailable)
+   * @param expectedDeviceId Not yet implemented (Unavailable)
+   * @return Not yet implemented (Unavailable)
+   */
+  public static boolean cloudRemoteAccessConnect_530(
+      CConnectorMqttMgr mqttMgr, String mqttTopic, String message, String expectedDeviceId) {
+    return false;
+  }
+
+  /**
+   * Downloads the firmware file from the specified remote URL to the specified local file.
+   *
+   * @param remoteFileUrl The URL of the remote file to download the firmware file from.
+   * @return A string indicating the reason for download failure, otherwise, null if successful.
+   * @throws EWException if the download fails, see the Ewon event logs.
+   */
+  private static String downloadFirmwareFile(String remoteFileUrl)
+      throws EWException, MalformedURLException, JSONException {
+    // Perform GET request to specified URL
+    final String urlAuthNotEncoded =
+        CConnectorMain.getConnectorConfig().getCumulocityDeviceTenant()
+            + "/"
+            + CConnectorMain.getConnectorConfig().getCumulocityDeviceUsername()
+            + ":"
+            + CConnectorMain.getConnectorConfig().getCumulocityDevicePassword();
+    final String urlAuthEncoded = Base64.encodeBytes(urlAuthNotEncoded.getBytes());
+    final String authHeader = "Authorization=Basic " + urlAuthEncoded;
+    int httpStatus =
+        ScheduledActionManager.RequestHttpX(
+            remoteFileUrl, "GET", authHeader, "", "", "/ewonfwr.edf");
+
+    // Read response contents and return
+    String resultString = null;
+    if (httpStatus == SCHttpUtility.HTTPX_CODE_EWON_ERROR) {
```

Comment:
I think this is simpler as an if statement. It is shorter, and using a switch statement would mean we need to introduce an additional case, because there is no clear _default_ case (no else statement)

Reply:
The last `else if` in this statement has a != comparison against the no_error value, thus unknown errors are already handled:

![image](https://user-images.githubusercontent.com/51409174/155748612-8b0a483c-f227-4c6a-86a4-06fd3700068d.png)

Reply:
In this scenario, I believe that a switch statement is only going to make this more confusing by introducing nested comparisons. Switch statements do not support the != operator, so a switch statement here would mean we would need a default case, and then an if statement inside the default case that still is checking `if (httpStatus != SCHttpUtility.HTTPX_CODE_NO_ERROR)`.

Reply:
say what? 
case httpx_node_no_error: 
  // not an error 
 break;
default: 
// all undocumented error codes 
break;

Reply:
Using an if statement here results in 20 lines of code, while a switch statement is 27 lines.

In the end though, I think they are functionally the same and it really might just come down to developer opinion.